server/server.go

```diff
@@ -642,3 +660,45 @@ func checkBranchTranslationCmdF(args []string, w http.ResponseWriter, slashComma
 
 	return nil
 }
+
+func pipelineTriggerCmdF(args []string, w http.ResponseWriter, slashCommand *MMSlashCommand) error {
+	const colorErr = "#ee2116"
+	const colorSuccess = "#0060aa"
+
+	if len(args) < 1 {
+		msg := "You need to set at least one pipeline name to trigger."
+		if len(Cfg.PipelineTriggers) == 0 {
+			msg += "No trigger is configured. Please configure!"
+		} else {
+			msg += "Configured Pipelines:"
+			for name, trigger := range Cfg.PipelineTriggers {
+				msg += fmt.Sprintf("\n**%s**: %s", name, trigger.Description)
+			}
+		}
+		WriteEnrichedResponse(w, "Trigger Pipeline", msg, colorErr, model.COMMAND_RESPONSE_TYPE_IN_CHANNEL)
+		return nil
+	}
+
+	triggerName := args[0]
+	pipelineTrigger, ok := Cfg.PipelineTriggers[triggerName]
+
+	if !ok {
+		WriteEnrichedResponse(w, "Trigger Pipeline", fmt.Sprintf("%s is not defined!", triggerName), colorErr, model.COMMAND_RESPONSE_TYPE_IN_CHANNEL)
+		return nil
+	}
+
+	uid, ok := pipelineTrigger.Users[slashCommand.Username]
+	if !ok || uid != slashCommand.UserID {
+		WriteEnrichedResponse(w, "Trigger Pipeline", fmt.Sprintf("You are not allowed to trigger %s pipeline", triggerName), colorErr, model.COMMAND_RESPONSE_TYPE_IN_CHANNEL)
+		return nil
+	}
+
+	pipelineURL, err := TriggerPipeline(pipelineTrigger, args)
+	if err != nil {
+		return err
```

Comment:
Should we write a response when the trigger fails too?

Reply:
Send response to the user including error details.
https://github.com/mattermost/matterbuild/pull/92/files#diff-78f42ba40d0f10b08c73b7e6bb8376f398e249c963cf549e89591d6b6826b9a4R697-R699